### PR TITLE
{2023.06}[foss/2023a] CP2K 2023.1

### PR DESCRIPTION
**Pull request by laraPPr**

### CODE REVIEW

**Comment by laraPPr** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic

**Comment by laraPPr** [comment]
log of aarch64/generic:
```
================================================================================

LIBXSMM release-1.17 (Linux@aarch64-generic-node1.int.aws-rocky88-202310.eessi.io)

--------------------------------------------------------------------------------

GNU Compiler Collection: gcc 12.3.0, g++ 12.3.0, and gfortran 12.3.0

C / C++ target: <compiler default>

Fortran target: <compiler default>

--------------------------------------------------------------------------------

Environment: CC CXX FC

--------------------------------------------------------------------------------

--- LIBXSMM build log

gcc  -DNDEBUG -D__STATIC=1 -DLIBXSMM_BUILD=2 -Iinclude -I./src -Iobj  -fPIC -Wall -O2 -fdata-sections -ffunction-sections -fvisibility=hidden -pthread  -c ./src/libxsmm_main.c -o  obj/intel64/libxsmm_main.o

In file included from ./src/libxsmm_trace.h:14,

                 from ./src/libxsmm_main.c:11:

include/libxsmm_macros.h:61:5: error: #error Intel Architecture or compatible CPU required!

   61 | #   error Intel Architecture or compatible CPU required!

      |     ^~~~~

./src/libxsmm_main.c: In function internal_init:

./src/libxsmm_main.c:866:3: warning: pthread_yield is deprecated: pthread_yield is deprecated, use sched_yield instead [-Wdeprecated-declarations]

  866 |   LIBXSMM_LOCK_ACQUIRE(LIBXSMM_LOCK, &libxsmm_lock_global);

      |   ^~~~~~~~~~~~~~~~~~~~

In file included from /cvmfs/software.eessi.io/versions/2023.06/compat/linux/aarch64/usr/include/features.h:491,

                 from /cvmfs/software.eessi.io/versions/2023.06/compat/linux/aarch64/usr/include/pthread.h:21,

                 from include/libxsmm_macros.h:815:

/cvmfs/software.eessi.io/versions/2023.06/compat/linux/aarch64/usr/include/pthread.h:479:12: note: declared here

  479 | extern int __REDIRECT_NTH (pthread_yield, (void), sched_yield)

      |            ^~~~~~~~~~~~~~

./src/libxsmm_main.c: In function libxsmm_init:

./src/libxsmm_main.c:1224:7: warning: pthread_yield is deprecated: pthread_yield is deprecated, use sched_yield instead [-Wdeprecated-declarations]

 1224 |       while (2 > LIBXSMM_ATOMIC_LOAD(&libxsmm_ninit, LIBXSMM_ATOMIC_RELAXED)) LIBXSMM_SYNC_YIELD;

      |       ^~~~~

/cvmfs/software.eessi.io/versions/2023.06/compat/linux/aarch64/usr/include/pthread.h:479:12: note: declared here

  479 | extern int __REDIRECT_NTH (pthread_yield, (void), sched_yield)

      |            ^~~~~~~~~~~~~~

./src/libxsmm_main.c: In function libxsmm_finalize:

./src/libxsmm_main.c:1241:5: warning: pthread_yield is deprecated: pthread_yield is deprecated, use sched_yield instead [-Wdeprecated-declarations]

 1241 |     LIBXSMM_LOCK_ACQUIRE(LIBXSMM_LOCK, &libxsmm_lock_global);

      |     ^~~~~~~~~~~~~~~~~~~~

/cvmfs/software.eessi.io/versions/2023.06/compat/linux/aarch64/usr/include/pthread.h:479:12: note: declared here

  479 | extern int __REDIRECT_NTH (pthread_yield, (void), sched_yield)

      |            ^~~~~~~~~~~~~~

make: [Makefile:794: obj/intel64/libxsmm_main.o] Error 1 (ignored)

make: *** [Makefile:795: obj/intel64/libxsmm_main.o] Error 1
```

**Comment by laraPPr** [comment]
Also found this which shows that this is an expected error for ARM https://www.cp2k.org/dev:compiler_support

**Comment by bedroge** [comment]
The error seems related to libxsmm, see https://github.com/libxsmm/libxsmm/blob/1.17/include/libxsmm_macros.h#L61.

The main branch does seem to support Arm: https://github.com/libxsmm/libxsmm/blob/main/include/libxsmm_macros.h.

Version 1.17 is 3 years old, but they haven't released a new version...

**Comment by laraPPr** [comment]
Had a quick look at the easyblock and it looks like libxsmm can be removed from the depedencies for CP2K and thand the `-D__LIBXSMM` flag is not set. SO I can try that

**Comment by laraPPr** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic

**Comment by laraPPr** [comment]
@Bob @casparvl @boegel Do any of you have access to the logs of the bot to see what went wrong with https://github.com/EESSI/software-layer/pull/630#issuecomment-2202686272. Because is seems like the build of CP2K on Aarch64 succeeded (there is also no log for this job on the shared folder in the magic castle cluster). But now I'm wondering what did go wrong. And why did this not happen in https://github.com/EESSI/software-layer/pull/472 where I had done the exact same thing as I now did for CP2K.

**Comment by bedroge** [comment]
> @bedroge @casparvl @boegel Do any of you have access to the logs of the bot to see what went wrong with [#630 (comment)](https://github.com/EESSI/software-layer/pull/630#issuecomment-2202686272). Because is seems like the build of CP2K on Aarch64 succeeded (there is also no log for this job on the shared folder in the magic castle cluster). But now I'm wondering what did go wrong. And why did this not happen in #472 where I had done the exact same thing as I now did for CP2K.

@laraPPr I checked the log and the build actually completed successfully. However, the missing installation check at the end of the installation script is still complaining that libxsmm is missing. Perhaps we need a similar hook as the one you've added for LAMMPS here https://github.com/EESSI/software-layer/pull/472/files#diff-03a1681f29b92037b3b19f52498d8a045fb14dcdebbaaadee3ffaaaa2dcd2ea7?

**Comment by laraPPr** [comment]
The check missing installations was only added 4 months ago (See https://github.com/EESSI/software-layer/blame/a581720b38efbe3302154c82a570d0e8f26d2ad6/EESSI-install-software.sh#L292). That's why it did not show up for lammps yet I'll add the hook

**Comment by laraPPr** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic

**Comment by laraPPr** [comment]
> > @bedroge @casparvl @boegel Do any of you have access to the logs of the bot to see what went wrong with [#630 (comment)](https://github.com/EESSI/software-layer/pull/630#issuecomment-2202686272). Because is seems like the build of CP2K on Aarch64 succeeded (there is also no log for this job on the shared folder in the magic castle cluster). But now I'm wondering what did go wrong. And why did this not happen in #472 where I had done the exact same thing as I now did for CP2K.
> 
> @laraPPr I checked the log and the build actually completed successfully. However, the missing installation check at the end of the installation script is still complaining that libxsmm is missing. Perhaps we need a similar hook as the one you've added for LAMMPS here https://github.com/EESSI/software-layer/pull/472/files#diff-03a1681f29b92037b3b19f52498d8a045fb14dcdebbaaadee3ffaaaa2dcd2ea7?

The `check_missing_installations.sh` was not in place yet when LAMMPS was merged I think so its gonna need something extra in the hook because it is still failing, see https://github.com/EESSI/software-layer/pull/630#issuecomment-2202686272.

**Comment by laraPPr** [comment]
The parse hook removes the dependencies but it still picks up the dependencies of these dependencies as missing check with the LAMMPS installation in EESSI: 
```
{EESSI 2023.06} [larappr@aarch64-generic-node1 ~]$ eb LAMMPS-2Aug2023_update2-foss-2023a-kokkos.eb -M --hooks=eb_hooks.py 

== Temporary log file in case of crash /tmp/eb-cyvd969m/easybuild-8jp9w0jy.log

== found valid index for /cvmfs/software.eessi.io/versions/2023.06/software/linux/aarch64/neoverse_n1/software/EasyBuild/4.9.2/easybuild/easyconfigs, so using it...

== found valid index for /cvmfs/software.eessi.io/versions/2023.06/software/linux/aarch64/neoverse_n1/software/EasyBuild/4.9.2/easybuild/easyconfigs, so using it...

== Using custom configure options for UCX: --enable-optimizations --enable-cma --enable-mt --with-verbs --without-java --without-go --disable-doxygen-doc --disable-logging --disable-debug --disable-assertions --disable-params-check  --with-sysroot=/cvmfs/software.eessi.io/versions/2023.06/compat/linux/aarch64  --with-rdmacm=/cvmfs/software.eessi.io/versions/2023.06/compat/linux/aarch64/usr 

== Added '--with-add-fonts=/cvmfs/software.eessi.io/versions/2023.06/compat/linux/aarch64/usr/share/fonts' configure option for fontconfig



27 out of 124 required modules missing:



* M4/1.4.19 (M4-1.4.19.eb)

* Bison/3.8.2 (Bison-3.8.2.eb)

* flex/2.6.4 (flex-2.6.4.eb)

* zlib/1.2.13 (zlib-1.2.13.eb)

* binutils/2.40 (binutils-2.40.eb)

* zlib/1.2.13-GCCcore-12.3.0 (zlib-1.2.13-GCCcore-12.3.0.eb)

* help2man/1.49.3-GCCcore-12.3.0 (help2man-1.49.3-GCCcore-12.3.0.eb)

* M4/1.4.19-GCCcore-12.3.0 (M4-1.4.19-GCCcore-12.3.0.eb)

* flex/2.6.4-GCCcore-12.3.0 (flex-2.6.4-GCCcore-12.3.0.eb)

* binutils/2.40-GCCcore-12.3.0 (binutils-2.40-GCCcore-12.3.0.eb)

* bzip2/1.0.8-GCCcore-12.3.0 (bzip2-1.0.8-GCCcore-12.3.0.eb)

* ncurses/6.4-GCCcore-12.3.0 (ncurses-6.4-GCCcore-12.3.0.eb)

* libreadline/8.2-GCCcore-12.3.0 (libreadline-8.2-GCCcore-12.3.0.eb)

* ncurses/6.3 (ncurses-6.3.eb)

* gettext/0.21.1 (gettext-0.21.1.eb)

* Autoconf/2.71-GCCcore-12.3.0 (Autoconf-2.71-GCCcore-12.3.0.eb)

* Automake/1.16.5-GCCcore-12.3.0 (Automake-1.16.5-GCCcore-12.3.0.eb)

* XZ/5.4.2-GCCcore-12.3.0 (XZ-5.4.2-GCCcore-12.3.0.eb)

* libtool/2.4.7-GCCcore-12.3.0 (libtool-2.4.7-GCCcore-12.3.0.eb)

* Autotools/20220317-GCCcore-12.3.0 (Autotools-20220317-GCCcore-12.3.0.eb)

* Yasm/1.3.0-GCCcore-12.3.0 (Yasm-1.3.0-GCCcore-12.3.0.eb)

* gettext/0.21.1-GCCcore-12.3.0 (gettext-0.21.1-GCCcore-12.3.0.eb)

* gperf/3.1-GCCcore-12.3.0 (gperf-3.1-GCCcore-12.3.0.eb)

* groff/1.22.4-GCCcore-12.3.0 (groff-1.22.4-GCCcore-12.3.0.eb)

* Perl-bundle-CPAN/5.36.1-GCCcore-12.3.0 (Perl-bundle-CPAN-5.36.1-GCCcore-12.3.0.eb)

* intltool/0.51.0-GCCcore-12.3.0 (intltool-0.51.0-GCCcore-12.3.0.eb)

* util-linux/2.39-GCCcore-12.3.0 (util-linux-2.39-GCCcore-12.3.0.eb)



== Temporary log file(s) /tmp/eb-cyvd969m/easybuild-8jp9w0jy.log* have been removed.

== Temporary directory /tmp/eb-cyvd969m has been removed. 
```
So LAMMPS would also no longer pass with the `check_missing_installations.sh` now in place
As you can see `tbb` and `ScaFaCos` are not in the list of missing installations but their dependencies are so I also do not know how this passed the CI in Febuary and probably will not anymore now.

**Comment by bedroge** [comment]
@laraPPr I see a lot of dependencies that should be filtered, see https://github.com/EESSI/software-layer/blob/2023.06-software.eessi.io/configure_easybuild#L28. Did you source the `configure_easybuild` script?

**Comment by laraPPr** [comment]
> @laraPPr I see a lot of dependencies that should be filtered, see https://github.com/EESSI/software-layer/blob/2023.06-software.eessi.io/configure_easybuild#L28. Did you source the `configure_easybuild` script?

That did the trick for checking lammps but now I still don't know what is missing for CP2K

**Comment by laraPPr** [comment]
@bedroge can you check the logs for it might not have printed `No missing installations`
for this job https://github.com/EESSI/software-layer/pull/630#issuecomment-2205679778?

**Comment by laraPPr** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic inst:eessi-bot-mc-aws

**Comment by laraPPr** [comment]
It was a little parsing mistake in the hook that was not caught in the CI fix it and now build for aarch64/generic passed

**Comment by laraPPr** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1

bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by laraPPr** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic 
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell 
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512 
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2 
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3 
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1 
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by laraPPr** [comment]
Since they all failed even the ones that passed before something fishy happened. I'll have a look at the logs

**Comment by laraPPr** [comment]
Ah it failed to create an artefact for all of them I'll kick some of again and see if it is resolved. but is a little strange

**Comment by laraPPr** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512

**Comment by laraPPr** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512

**Comment by boegel** [comment]
> Ah it failed to create an artefact for all of them I'll kick some of again and see if it is resolved. but is a little strange

Probably fallout due to the bugs being fixed in #635 and #636, should be OK now

**Comment by laraPPr** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by laraPPr** [comment]
Could the fact that no artefact was created have something to do with this, https://github.com/EESSI/software-layer/pull/630#issuecomment-2212490549?

File: easystacks/software.eessi.io/2023.06/eessi-2023.06-eb-4.9.2-2023a.yml [hunk 1]
```diff
@@ -16,3 +16,7 @@ easyconfigs:
   - MAFFT-7.520-GCC-12.3.0-with-extensions.eb
   - ncbi-vdb-3.0.10-gompi-2023a.eb
   - MetalWalls-21.06.1-foss-2023a.eb
+  - CP2K-2023.1-foss-2023a.eb:
+      # see https://github.com/easybuilders/easybuild-easyconfigs/pull/20951
+      options:
+        from-pr: 20951
```
Comment (bedroge) [inline easystacks/software.eessi.io/2023.06/eessi-2023.06-eb-4.9.2-2023a.yml:22]:
```suggestion
  - CP2K-2023.1-foss-2023a.eb:
      options:
        # see https://github.com/easybuilders/easybuild-easyconfigs/pull/20951
        from-commit: a92667fe32396bbd4106243658625f7ff2adcd68
```

Comment (ocaisa) [reply easystacks/software.eessi.io/2023.06/eessi-2023.06-eb-4.9.2-2023a.yml:22]: @laraPPr is on holidays so I will commit this suggestion

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen4

**Comment by ocaisa** [comment]
I'm trying a zen4 build as well

**Review by bedroge — APPROVED** [review]

**Comment by bedroge** [comment]
Almost every tarball seems to include a different number of files. Except for the x86_64 vs aarch64 difference w.r.t. libxsmm, I don't see any obvious differences/causes, so I'm just going to assume this is normal...

**Pull request merged**